src/ProgressOnderwijsUtils/Win32/VirusScan.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using Microsoft.Windows.Sdk;
+
+namespace ProgressOnderwijsUtils.Win32
+{
+    public static class VirusScan
+    {
+        public static unsafe bool IsMalware(byte[] buffer, string contentName, string sessionName)
+        {
+            var context = default(nint);
+
+            try {
+                PInvoke.AmsiInitialize(sessionName, out context).AssertResultOk();
+
+                fixed (void* bufferPtr = buffer) {
+                    PInvoke.AmsiScanBuffer(context, bufferPtr, (uint)buffer.LongLength, contentName, IntPtr.Zero, out var result).AssertResultOk();
+                    return ResultIsMalware(result);
+                }
+            } finally {
+                if (context != default) {

```

Comment:
Eamon is tegen naked default.

Reply:
R# vond het wel beter, maar vooruit

Reply:
`IntPtr.Zero`had vast ook gewerkt.

Reply:
er is `null` en `new()` als alternatief; die hebben het voordeel dat direct duidelijk is of iets nullable is of niet.